src/modules/lua_mtev_stats.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2017, Circonus, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ *       copyright notice, this list of conditions and the following
+ *       disclaimer in the documentation and/or other materials provided
+ *       with the distribution.
+ *     * Neither the name Circonus, Inc. nor the names of its contributors
+ *       may be used to endorse or promote products derived from this
+ *       software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "mtev_defines.h"
+
+#include "circmetrics.h"
+
+#include "mtev_conf.h"
+
+#include "lua_mtev.h"
+
+#define PUSH_OBJ(L, tname, obj) do { \
+  *(void **)(lua_newuserdata(L, sizeof(void *))) = (obj); \
+  luaL_getmetatable(L, tname); \
+  lua_setmetatable(L, -2); \
+} while(0)
+
+static int
+mtev_lua_stats_handle_record(lua_State *L) {
+  void **udata;
+  stats_handle_t *handle;
+  stats_type_t type;
+  int nargs = lua_gettop(L);
+  if(nargs < 2)
+    luaL_error(L, "libcircmetrics.handle.record(self,value[,cnt])");
+  udata = lua_touserdata(L, lua_upvalueindex(1));
+  if(udata != lua_touserdata(L, 1)) luaL_error(L, "must be called as method");
+  handle = *udata;
+  type = stats_handle_type(handle);
+  if(lua_isnil(L,2)) {
+    /* This clears a value */
+    stats_set(handle, type, NULL);
+    return 0;
+  }
+  switch(type) {
+    case STATS_TYPE_STRING:
+      stats_set(handle, STATS_TYPE_STRING, (char *)lua_tostring(L,2));
```

Comment:
recommend verifying that argument is a string before calling `stats_set`. similar comment applies to all numeric types.

Reply:
This was actually intentional as it seems more lua-esque to auto convert when possible like other functions do.  It allos setting strings like 1234 and having them be "1234" and someone setting a "9.2" number and having it flow through as 9.2.